receiver/jaegerreceiver/factory.go

```diff
@@ -59,33 +64,49 @@ func (f *Factory) Type() string {
 	return typeStr
 }
 
-// CustomUnmarshaler returns nil because we don't need custom unmarshaling for this config.
+// CustomUnmarshaler is used to add defaults for named but empty protocols
 func (f *Factory) CustomUnmarshaler() receiver.CustomUnmarshaler {
-	return nil
+	return func(v *viper.Viper, viperKey string, intoCfg interface{}) error {
+		// first load the config normally
+		vReceiverCfg := config.GetConfigSection(v, viperKey)
```

Comment:
I thought a bit more about this and there is an alternate solution.

Instead of making GetConfigSection exported and calling it here for the second time, what if we simply pass to CustomUnmarshaler one more parameter, the viper section that was already fetched by the earlier call to GetConfigSection in loadReceivers, and name the parameter e.g. `sourceViperSection *viper.Viper`?

This way there is no longer need to export GetConfigSection, don't need to call it the second time and do it in a way that matches the behavior of loadReceivers, which is and internal and may be changed in the future in a way that we cannot predict now.

What do you think?

Reply:
Yup, works for me.

I had considered something like this since `sv` is sitting right there in `loadReceivers()` and is exactly what I needed.  Will push the fix shortly.